### ISSUE TITLE
Remembered Device name missing

### ISSUE THREAD

**Issue by karolinafelus** [aws-amplify/amplify-js#12854]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React

### Amplify APIs

Authentication

### Amplify Version

v6

### Amplify Categories

auth

### Backend

None

### Environment information

<details>

```
# Put output below this line
  System:
    OS: Windows 11 10.0.22621
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz
    Memory: 20.18 GB / 31.61 GB
  Binaries:
    Node: 20.10.0 - C:\Program Files\nodejs\node.EXE
    npm: 10.2.5 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Chromium (120.0.2210.91)
    Internet Explorer: 11.0.22621.1
  npmPackages:
    @ant-design/icons: ^5.1.4 => 5.2.6
    @tanstack/eslint-plugin-query:  5.10.0
    @tanstack/query-async-storage-persister:  5.10.0
    @tanstack/query-broadcast-client-experimental:  5.10.0
    @tanstack/query-codemods:  4.24.3
    @tanstack/query-core:  5.10.0
    @tanstack/query-devtools:  5.10.0
    @tanstack/query-example-nextjs-suspense-streaming:  undefined ()  
    @tanstack/query-example-react-algolia:  undefined ()
    @tanstack/query-example-react-basic:  undefined ()
    @tanstack/query-example-react-basic-graphql-request:  undefined ()
    @tanstack/query-example-react-basic-typescript:  undefined ()     
    @tanstack/query-example-react-default-query-function:  undefined ()
    @tanstack/query-example-react-infinite-query-with-max-pages:  undefined ()
    @tanstack/query-example-react-load-more-infinite-scroll:  undefined ()
    @tanstack/query-example-react-nextjs:  undefined ()
    @tanstack/query-example-react-offline:  undefined ()
    @tanstack/query-example-react-optimistic-updates-cache:  undefined ()
    @tanstack/query-example-react-optimistic-updates-ui:  undefined ()
    @tanstack/query-example-react-pagination:  undefined ()
    @tanstack/query-example-react-playground:  undefined ()
    @tanstack/query-example-react-prefetching:  undefined ()
    @tanstack/query-example-react-react-native:  undefined ()
    @tanstack/query-example-react-refetch-interval:  undefined ()     
    @tanstack/query-example-react-rick-morty:  undefined ()
    @tanstack/query-example-react-router:  undefined ()
    @tanstack/query-example-react-simple:  undefined ()
    @tanstack/query-example-react-star-wars:  undefined ()
    @tanstack/query-example-react-suspense:  undefined ()
    @tanstack/query-example-solid-basic-graphql-request:  undefined ()
    @tanstack/query-example-solid-basic-typescript:  undefined ()     
    @tanstack/query-example-solid-default-query-function:  undefined ()
    @tanstack/query-example-solid-simple:  undefined ()
    @tanstack/query-example-solid-solid-start-streaming:  undefined ()
    @tanstack/query-example-svelte-auto-refetching:  undefined ()     
    @tanstack/query-example-svelte-basic:  undefined ()
    @tanstack/query-example-svelte-load-more-infinite-scroll:  undefined ()
    @tanstack/query-example-svelte-optimistic-updates-typescript:  undefined ()
    @tanstack/query-example-svelte-playground:  undefined ()
    @tanstack/query-example-svelte-simple:  undefined ()
    @tanstack/query-example-svelte-ssr:  undefined ()
    @tanstack/query-example-svelte-star-wars:  undefined ()
    @tanstack/query-example-vue-2.6-basic:  undefined ()
    @tanstack/query-example-vue-basic:  undefined ()
    @tanstack/query-example-vue-dependent-queries:  undefined ()      
    @tanstack/query-example-vue-nuxt3:  undefined ()
    @tanstack/query-example-vue-persister:  undefined ()
    @tanstack/query-persist-client-core:  5.10.0
    @tanstack/query-sync-storage-persister:  5.10.0
    @tanstack/react-query: ^5.0.5 => 5.10.0
    @tanstack/react-query-devtools: ^5.4.3 => 5.10.0
    @tanstack/react-query-next-experimental:  5.10.0
    @tanstack/react-query-persist-client:  5.10.0
    @tanstack/solid-query:  5.10.0
    @tanstack/solid-query-devtools:  5.10.0
    @tanstack/solid-query-persist-client:  5.10.0
    @tanstack/svelte-query:  5.10.0
    @tanstack/svelte-query-devtools:  5.10.0
    @tanstack/svelte-query-persist-client:  5.10.0
    @tanstack/vue-query:  5.10.0
    @tanstack/vue-query-devtools:  5.10.0
    @types/node: ^20.5.9 => 20.10.1
    @types/react: ^18.0.37 => 18.2.39
    @types/react-dom: ^18.0.11 => 18.2.17
    @types/react-router-dom: ^5.3.3 => 5.3.3
    @types/uuid: ^9.0.7 => 9.0.7
    @typescript-eslint/eslint-plugin: ^5.59.0 => 5.62.0
    @typescript-eslint/parser: ^5.59.0 => 5.62.0
    @vitejs/plugin-react: ^4.0.0 => 4.2.0
    amazon-cognito-identity-js: ^6.3.1 => 6.3.9
    amazon-cognito-identity-js/internals:  undefined ()
    antd: ^5.6.3 => 5.11.5
    aws-amplify: ^6.0.10 => 6.0.10
    aws-amplify/adapter-core:  undefined ()
    aws-amplify/analytics:  undefined ()
    aws-amplify/analytics/kinesis:  undefined ()
    aws-amplify/analytics/kinesis-firehose:  undefined ()
    aws-amplify/analytics/personalize:  undefined ()
    aws-amplify/analytics/pinpoint:  undefined ()
    aws-amplify/api:  undefined ()
    aws-amplify/api/server:  undefined ()
    aws-amplify/auth:  undefined ()
    aws-amplify/auth/cognito:  undefined ()
    aws-amplify/auth/cognito/server:  undefined ()
    aws-amplify/auth/enable-oauth-listener:  undefined ()
    aws-amplify/auth/server:  undefined ()
    aws-amplify/datastore:  undefined ()
    aws-amplify/in-app-messaging:  undefined ()
    aws-amplify/in-app-messaging/pinpoint:  undefined ()
    aws-amplify/push-notifications:  undefined ()
    aws-amplify/push-notifications/pinpoint:  undefined ()
    aws-amplify/storage:  undefined ()
    aws-amplify/storage/s3:  undefined ()
    aws-amplify/storage/s3/server:  undefined ()
    aws-amplify/storage/server:  undefined ()
    aws-amplify/utils:  undefined ()
    aws-sdk: ^2.1414.0 => 2.1509.0
    axios: ^1.4.0 => 1.6.2
    chart.js: ^4.4.0 => 4.4.0
    chart.js-auto:  undefined ()
    chart.js-helpers:  undefined ()
    dec-to-binary: ^1.5.1 => 1.5.1
    eslint: ^8.38.0 => 8.54.0
    eslint-plugin-react-hooks: ^4.6.0 => 4.6.0
    eslint-plugin-react-refresh: ^0.3.4 => 0.3.5
    jszip: ^3.10.1 => 3.10.1
    qrcode.react: ^3.1.0 => 3.1.0
    query: github:tanstack/react-query => undefined ()
    react: ^18.2.0 => 18.2.0
    react-chartjs-2: ^5.2.0 => 5.2.0
    react-cra4:  undefined ()
    react-cra5:  undefined ()
    react-dnd: ^16.0.1 => 16.0.1
    react-dnd-html5-backend: ^16.0.1 => 16.0.1
    react-dnd-touch-backend: ^16.0.1 => 16.0.1
    react-dom: ^18.2.0 => 18.2.0
    react-next:  undefined ()
    react-query: ^3.39.3 => 3.39.3
    react-router-dom: ^6.14.1 => 6.20.0
    react-vite4:  undefined ()
    react-vite5:  undefined ()
    sass: ^1.63.6 => 1.69.5
    solid-vite:  undefined ()
    svelte-vite:  undefined ()
    typescript: ^5.0.2 => 5.3.2
    uuid: ^9.0.0 => 9.0.1 (8.3.2, 8.0.0)
    vite: ^4.3.9 => 4.5.0
    vue-vite:  undefined ()
    zod: ^3.22.2 => 3.22.4
    zustand: ^4.3.8 => 4.4.7
  npmGlobalPackages:
    npm: 10.2.5

```

</details>

### Describe the bug

After migrating my Auth code from Amplify v5 to v6, tracked devices name are not remembered in the Cognito console, which, obviously, prevents rememberDevice function to fetch them with a name, I receive only an ID. While I was using v5, the device name consisted of user agent data. Now, after User authenticates themselves, this column in Cognito UI is empty.
![cognito-gui-device-tracking](https://github.com/aws-amplify/amplify-js/assets/45790775/fb5aa0e6-7ae7-48b8-9b7d-5fd2b42b7421)

### Expected behavior

I expected the tracked device to have a name consisting of user agent data, like in v5.

### Reproduction steps

1. Sign in as a user in the app that uses Cognito.
2. Go into your Cognito UI.
3. Find the user from the first step in your User Pool.
4. Check the "Device tracking log" section - the names are empty.

### Code Snippet

```javascript
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

### aws-exports.js

Amplify.configure({
  Auth: {
    Cognito: {
    userPoolId: ap-northeast-1_XXXXXXXXX,
    userPoolClientId: XXXXXXXXXXXXXXXXXXXXXXXXX,
    }
  },
});

### Additional information and screenshots

I have written about it on Amplify discord and was instructed by developer (reesscot) to create an issue on github.
I do not know if I am missing something. Thank you for your help in advance.

**Comment by cwomack** [comment]
Hello, @karolinafelus and thank you for opening this issue.  This does appear to be a regression from the v5 to v6 functionality with the changes made to the `rememberDevice` API within Auth.  Marking this as a bug and we'll update this issue with progress as we make it.

**Comment by karolinafelus** [comment]
@cwomack I have a feature that allows my users to remove remembered device and I want my user to able to identify which device they're forgetting through at least user agent data.

**Comment by mewtlu** [comment]
This is key functionality for us which has regressed and gone undocumented in the migration to v6, could we get a status update on when this will be resolved please?

**Comment by beikeni** [comment]
Also being affected by this. I just completed the migration and will have to remove the feature for now.

**Comment by sohibegit** [comment]
We are using version 6.0.20. Although friendlyDeviceName is available in the SDK, it seems to be ignored; setting it to any value does not affect the output of fetchDevices(), as the results do not reflect our settings.

**Comment by cwomack** [comment]
Just wanted to update anyone following this issue that PR #13022 is in progress to address this.

**Comment by ashwinkumar6** [comment]
Hi @sohibegit, @beikeni, @mewtlu @karolinafelus 
This has been released in [v6.3.0](https://github.com/aws-amplify/amplify-js/releases/tag/aws-amplify%406.3.0). 

We add a default deviceName when MFA and Device tracking are enabled on Cognito. This deviceName is visible on the Cognito console as well as via the `fetchDevices` API ('name' property). 

Please let us know if you're run into any other issues!
